app/src/test/java/org/mozilla/fenix/ext/BookmarkNodeTest.kt

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ext
+
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.TestApplication
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+import mozilla.components.concept.storage.BookmarkNode
+import mozilla.components.concept.storage.BookmarkNodeType
+
+@RunWith(RobolectricTestRunner::class)
+@Config(application = TestApplication::class)
+class BookmarkNodeTest {
+    val nodeType = BookmarkNodeType.ITEM
+    val bookMarkChild1 = newBookMarkNode("Child 1", 1, null)
+    val bookMarkChild2 = newBookMarkNode("Child 2", 2, null)
+    val bookMarkChild3 = newBookMarkNode("Child 3", 3, null)
+    val allChildren = listOf(bookMarkChild1, bookMarkChild2)
+    val subsetToSubtract = setOf(bookMarkChild1)
+    val expectedRemainingSubset = listOf(bookMarkChild2)
+    val setofAllChildren = setOf(bookMarkChild1, bookMarkChild2)
+    val setofNonChildren = setOf(bookMarkChild3)
+    val emptyChildrenList: List<BookmarkNode> = emptyList()
+    val emptyChildrenSet: Set<BookmarkNode> = emptySet()
+    val bookmarkNode = BookmarkNode(nodeType, "11", "00", 0, "Testing", "testing.url", allChildren)
+
+    @Test
+    fun `PASS WHEN minusing a sub set of children THEN the children subset is removed`() {
```

Comment:
nit: PASS is unnecessary: sorry, that was an example of what's printed when running the tests, e.g. via a command line interface. Just `GIVEN ... WHEN ... THEN`